src/read.c

```diff
@@ -760,6 +760,10 @@ static avifMeta * avifMetaCreate()
 
 static void avifMetaDestroy(avifMeta * meta)
 {
+    if (meta == NULL) {
+      return;
+    }
```

Comment:
Nit: The current style in src/read.c is to have the caller handle this. For example, see the second `avifMetaDestroy()` call in `avifDecoderDataDestroy()`:

```
static void avifDecoderDataDestroy(avifDecoderData * data)
{
    avifMetaDestroy(data->meta);
    for (uint32_t i = 0; i < data->tracks.count; ++i) {
        avifTrack * track = &data->tracks.track[i];
        if (track->sampleTable) {
            avifSampleTableDestroy(track->sampleTable);
        }
        if (track->meta) {
            avifMetaDestroy(track->meta);
        }
    }
    avifArrayDestroy(&data->tracks);
    avifDecoderDataClearTiles(data);
    avifArrayDestroy(&data->tiles);
    avifFree(data);
}
```

So I suggest adding a similar null pointer check to the first `avifMetaDestroy()` call in `avifDecoderDataDestroy()`.

```
    if (data->meta) {
        avifMetaDestroy(data->meta);
    }
```

Reply:
> Nit: The current style in src/read.c is to have the caller handle this. For example, see the second `avifMetaDestroy()` call in `avifDecoderDataDestroy()`:
> 
> ```
> static void avifDecoderDataDestroy(avifDecoderData * data)
> {
>     avifMetaDestroy(data->meta);
>     for (uint32_t i = 0; i < data->tracks.count; ++i) {
>         avifTrack * track = &data->tracks.track[i];
>         if (track->sampleTable) {
>             avifSampleTableDestroy(track->sampleTable);
>         }
>         if (track->meta) {
>             avifMetaDestroy(track->meta);
>         }
>     }
>     avifArrayDestroy(&data->tracks);
>     avifDecoderDataClearTiles(data);
>     avifArrayDestroy(&data->tiles);
>     avifFree(data);
> }
> ```
> 
> So I suggest adding a similar null pointer check to the first `avifMetaDestroy()` call in `avifDecoderDataDestroy()`.
> 
> ```
>     if (data->meta) {
>         avifMetaDestroy(data->meta);
>     }
> ```

Done